assets/js/extensions/jetpack/woocommerce-analytics/index.ts

```diff
@@ -0,0 +1,175 @@
+/**
+ * External dependencies
+ */
+import { Cart, isObject, objectHasProp } from '@woocommerce/types';
+import { select } from '@wordpress/data';
+import { getSetting } from '@woocommerce/settings';
+
+/**
+ * Internal dependencies
+ */
+import { STORE_KEY as CART_STORE_KEY } from '../../../data/cart/constants';
+
+declare global {
+	interface Window {
+		// eslint-disable-next-line @typescript-eslint/naming-convention
+		_wca: {
+			// eslint-disable-next-line @typescript-eslint/ban-types
+			push: ( properties: Record< string, unknown > ) => void;
+		};
+	}
+}
+
+/**
+ * Check if the _wca object is valid and has a push property that is a function.
+ *
+ * @param  wca {unknown} Object that might be a Jetpack WooCommerce Analytics object.
+ */
+// eslint-disable-next-line @typescript-eslint/ban-types
+const isValidWCA = (
+	wca: unknown
+): wca is { push: ( properties: Record< string, unknown > ) => void } => {
+	if ( ! isObject( wca ) || ! objectHasProp( wca, 'push' ) ) {
+		return false;
+	}
+	return typeof wca.push === 'function';
+};
+
+const registerActions = (): void => {
+	if ( ! isValidWCA( window._wca ) ) {
+		// eslint-disable-next-line no-useless-return
+		return;
+	}
+
+	// We will register actions here in a later PR.
+};
+
+document.addEventListener( 'DOMContentLoaded', () => {
+	registerActions();
+} );
+
```

Comment:
```suggestion
```

@senadir if you'd like to remove this code from this quickly PR feel free to apply this suggestion.